.github/workflows/ci-test.yml

```diff
@@ -42,6 +42,47 @@ jobs:
       if: always()
       run: docker-compose -f dist/ci/docker-compose.yml down
 
+  smoke-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - registry.opensuse.org/opensuse/tumbleweed:latest
+          - registry.opensuse.org/opensuse/leap:15.5
+
+    container: ${{ matrix.image }}
+    steps:
+      - name: install git to checkout the repo
+        run: zypper -n in git
+
+      - uses: actions/checkout@v3
+
+      - name: fix the file permissions of the repository
+        run: chown -R $(id -un):$(id -gn) .
+
+      - name: add the openSUSE:Tools repository for Tumbleweed
+        run: zypper -n ar https://download.opensuse.org/repositories/openSUSE:/Tools/openSUSE_Tumbleweed/openSUSE:Tools.repo
+        if: ${{ contains(matrix.image, 'tumbleweed') }}
```

Comment:
IMO `if` should be above `run`

Reply:
It doesn't matter where you put the `if`.

Reply:
Not semantically, but usually conditions are written before the code that they guard, to match the flow of reading and thought.

Reply:
:shrug: done